rest-service/manager_rest/storage/resource_models_base.py

```diff
@@ -137,3 +137,7 @@ def list_labels(labels):
                       created_at=label.created_at,
                       creator_id=label.creator.id)
                 for label in labels]
+
+    def check_unique_query(self):
+        if self.is_id_unique:
+            return self.__class__.query.filter(self.__class__.id == self.id)
```

Comment:
if other classes use `super()`, then this could apply the tenant filter, and sm's `_get_unique_resource_query` could be entirely removed

Reply:
also the tenant/visibility filter can be written as `self.__class__.query.tenant(instance.tenant).filter(self.__class__.id == self.id)`